service/history/hsm/tree.go

```diff
@@ -700,6 +730,48 @@ func (n *Node) root() *Node {
 	return root
 }
 
+// Compact filters the operation log to remove any transitions for deleted nodes while preserving deletion operations.
+// The returned log maintains deletion tombstones, but excludes transitions that occurred on deleted nodes or their
+// descendants.
+func (ol OperationLog) Compact() OperationLog {
+	var deletedPaths [][]Key
+	result := make(OperationLog, 0, len(ol))
+	hasDeleteOperation := make(map[string]bool)
```

Comment:
nit: use `map[something]struct{}` for sets in Go.

Reply:
Thx! Done